docs/style-spec/expressions.md

```diff
@@ -0,0 +1,173 @@
+
+# Style Functions
+
+The value for any layout or paint property may be specified as a function. Functions allow you to make the appearance of a map feature change with the current zoom level and/or the feature's properties.
+
+`expression`
+_Required [expression value](#Expressions)_
+An expression defines how one or more feature property values and/or the current zoom level are combined using logical, mathematical, string, or color operations to produce the appropriate style value.  See [Expressions](#Expressions) for syntax details.
+
+## Property functions
+
+<p>A <strong>property function</strong> is any function defined using an expression that includes a reference to `["properties"]`. Property functions allow the appearance of a map feature to change with its properties. They can be used to visually differentate types of features within the same layer or create data visualizations. Note that support for property functions is not available across all properties and platforms at this time.</p>
+
+```js
+{
+  "circle-color": {
+    "expression": [
+      'rgb',
+      // red is higher when feature.properties.temperature is higher
+      ["get", ["properties"], "temperature", "Number"],
+      0,
+      // blue is higher when feature.properties.temperature is lower
+      ["-", 100, ["get", ["properties"], "temperature", "Number"]]
+    ]
+  }
+}
+```
+
+
+## Zoom-dependent functions
+
+A <strong>zoom function</strong> is any function defined using an expression that includes a reference to `["zoom"]`.  Such functions allow the appearance of a map feature change with map’s zoom level. Zoom functions can be used to create the illusion of depth and control data density.  
+
+A zoom function must be of the following form (see [Curves](#Curves) below):
+
+```js
+{
+  "expression": [ "curve", interpolation, ["zoom"], ... ]
+}
+```
+
+(`["zoom"]` may not appear anywhere else in the expression.)
+
+
+### Example: a zoom-only function.
+
+```js
+{
+  "circle-radius": {
+    "expression": [
+      "curve", "linear", ["zoom"],
+      // zoom is 5 (or less) -> circle radius will be 1px
+      // zoom is 10 (or greater) -> circle radius will be 2px
+      5, 1, 10, 2
+    ]
+  }
+}
+```
+
+### Example: a zoom-and-property function
+
+Using property functions as the output value for one or more zoom stops allows
+the appearance of a map feature to change with both the zoom level _and_ the
+feature's properties.
+
+```js
+{
+  "circle-radius": {
+    "expression": [
+      "curve", "linear", ["zoom"],
+
+      // zoom is 0 and "rating" is 0 -> circle radius will be 0px
+      // zoom is 0 and "rating" is 5 -> circle radius will be 5px
+      0, [ "get", ["properties"], "rating", "Number" ],
+      // zoom is 20 and "rating" is 0 -> circle radius will be 4 * 0 = 0px
+      // zoom is 20 and "rating" is 5 -> circle radius will be 4 * 5 = 20px
+      10, [ "*", 4, [ "get", ["properties"], "rating", "Number" ] ]
+    ]
+  }
+}
+```
+
+## Property Expressions
```

Comment:
This document refers exclusively to “expressions”, but the title of this PR refers to “arbitrary expressions”. To avoid confusion, we should settle on one name for the feature and eliminate the other name. As catchy as “Arbex” sounds, I prefer simply “expressions” for these reasons:

* For Studio users unfamiliar with the style specification, “arbitrary” may not be perceived as a positive trait.
* “Arbitrary expressions” would seem to imply a non-arbitrary expression feature that doesn’t exist.
* “Expressions” aligns with common usage on the iOS platform and in other kinds of software.

/cc @dasulit